File: apps/api/tests/SafeSchool.Api.Tests/Features/AttendanceAccess/Fixtures/AttendanceAccessTestData.cs
using SafeSchool.Api.Features.AttendanceAccess.Common;
using SafeSchool.Api.Features.AttendanceAccess.Scans;

namespace SafeSchool.Api.Tests.Features.AttendanceAccess.Fixtures;

public static class AttendanceAccessTestData
{
    public static RecordScanRequest EntryScan(Guid gateId, Guid scanPointId, string credential = "credential-active", string clientScanId = "scan-1") =>
        new(gateId, scanPointId, credential, AttendanceDirection.Entry, ScanMethod.Nfc, clientScanId, DateTimeOffset.UtcNow.Date.AddHours(7).AddMinutes(45));

    public static CredentialEvidence ActiveCredential(string tenantId = "school-1", string student = "student-active", string credential = "credential-active") =>
        new(tenantId, student, credential, CredentialEvidenceStatus.Active);
}

File: apps/mobile/test/features/attendance_access/attendance_access_test_data.dart
import '../../../lib/features/attendance_access/scans/gate_scan_event.dart';

GateScanEvent testEntryScan({String clientScanId = 'scan-1'}) => GateScanEvent(
      clientScanId: clientScanId,
      gateId: 'gate-1',
      scanPointId: 'scan-point-1',
      credentialReference: 'credential-active',
      direction: AttendanceDirection.entry,
      method: ScanMethod.nfc,
      localScanTime: DateTime.utc(2026, 5, 4, 7, 45),
    );

